src/domain/IRegistrationGateway.ts

```diff
@@ -0,0 +1,15 @@
+import { LimitedPartnership } from "@companieshouse/api-sdk-node/dist/services/limited-partnerships";
+
+import PageRegistrationType from "../presentation/controller/registration/PageRegistrationType";
+
+interface IRegistrationGateway {
+  getSubmissionById(id: string): Promise<LimitedPartnership>;
+  createTransaction(pageType: PageRegistrationType): Promise<string>;
+  createTransactionAndFirstSubmission(
+    pageType: PageRegistrationType,
+    transactionId: string,
```

Comment:
because the service first creates the transaction and then calls this method to create the first submission by passing the transactionId
the name can be changed to createFirstSubmissionFromTransaction

Reply:
👍 now I understand.

Probably just `createSubmission` would be fine. Because a transaction id is being passed in, it's clear that this new submission will be linked to the transaction.